### PR TITLE
etcd backup

### LINKED ISSUE #1: Backup etcd
The [OpenShift 4 documentation](https://docs.openshift.com/container-platform/4.5/backup_and_restore/backing-up-etcd.html) explains how to backup etcd of a cluster.

* Make this procedure work within a container
* Make this component to configure K8up objects for backing up the etcd

### PR DESCRIPTION
**Pull request by bastjan**
Adds backup definitions for `etcd` on OpenShift 4.

Restore documentation will follow. Most likely as part of https://kb.vshn.ch/oc4/.

Fixes #1 .

## Checklist

- [ ] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.

### CODE REVIEW

File: class/defaults.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ parameters:
     # See: https://github.com/projectsyn/commodore/issues/71
     enabled: true
     namespace: syn-cluster-backup
+    etcd_backup_namespace: ${cluster_backup:namespace}-etcd
```
Comment (simu) [inline class/defaults.yml:8]: There's an argument to not even expose this parameter to users, and instead always use `'%s-etcd' % params.namespace` for the etcd backup namespace.

**Review by simu — CHANGES_REQUESTED** [review]
LGTM overall, some points that can be addressed to make the implementation a bit cleaner.

You may want to add a sentence explaining that the component now supports etcd backups on OCP4 in `docs/modules/ROOT/pages/index.adoc`.

Side-note: With the current implementation of the Prebackup Pod, we don't only backup the etcd snapshot but also the static pods which the `/usr/local/bin/cluster-backup.sh` script saves alongside the etcd snapshot. This behavior should be documented somewhere, or a link to relevant documentation should be added.

File: class/defaults.yml (hunk 1 above)
Comment (bastjan) [reply class/defaults.yml]: I think this is the way to go. There's no need to configure the name IMO.

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 2]
```diff
@@ -22,6 +22,14 @@ default:: `syn-cluster-backup`
...
 The namespace in which to deploy this component.
 
+== `etcd_backup_namespace`
+
+[horizontal]
+type:: string
+default:: `syn-cluster-backup-etcd`
+
+The namespace in which to deploy the etcd backup part of this component.
+
```
[the remark this answers is not in the document]
Comment (bastjan) [reply docs/modules/ROOT/pages/references/parameters.adoc]: Added the note to `docs/modules/ROOT/pages/index.adoc`

File: docs/modules/ROOT/pages/references/ocp4-etcd.adoc [hunk 3]
```diff
@@ -0,0 +1,12 @@
+= etcd Backup on Openshift 4
+
+etcd is the key-value store for OpenShift 4, which persists the state of all resource objects.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/ocp4-etcd.adoc:3]:
```suggestion
etcd is the key-value store used by Kubernetes.
etcd contains the state of all resources created in the cluster through the Kubernetes API.
```

File: docs/modules/ROOT/pages/references/ocp4-etcd.adoc [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+The automated backup follows the https://docs.openshift.com/container-platform/4.8/backup_and_restore/backing-up-etcd.html[Backing up etcd] guide.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/ocp4-etcd.adoc:5]:
```suggestion
The automated backup implements the steps documented in the https://docs.openshift.com/container-platform/4.8/backup_and_restore/backing-up-etcd.html[Backing up etcd] guide.
```

File: docs/modules/ROOT/pages/references/ocp4-etcd.adoc [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+== Files
+
+The backup tar contains two files:
```
Comment (simu) [inline docs/modules/ROOT/pages/references/ocp4-etcd.adoc:9]:
```suggestion
The backup consist of a `.tar.gz` archive which contains two files:
```

**Review by simu — APPROVED** [review]

**Pull request merged**